monitoring/src/websocket/ws_json_rpc.rs

```diff
@@ -58,7 +58,38 @@ pub async fn handle_request(
         );
 
         match Pin::from(fut).await {
-            Ok((_, resp)) => json_rpc_types::Response::result(req.jsonrpc, resp, req.id.clone()),
+            Ok(v) => {
+                let (_, body) = v.into_parts();
+                let body_bytes = match warp::hyper::body::to_bytes(body).await {
+                    Ok(body_bytes) => body_bytes,
+                    Err(_) => {
+                        let json_error = json_rpc_types::Error::with_custom_msg(
+                            json_rpc_types::ErrorCode::InternalError,
+                            "Failed to get bytes from hyper body",
+                        );
+                        return json_rpc_types::Response::error(
+                            req.jsonrpc,
+                            json_error,
+                            req.id.clone(),
+                        );
+                    }
+                };
+                let res = match serde_json::from_slice(&body_bytes) {
```

Comment:
@adonagy if you are always going to parse the already produced bytes here, cannot you just used `RawValue` to pass them as-is?

Reply:
iirc the previous hack was a workaround for serde not being able to properly parse a JSON value produced by the protocol, so we will have that issue again here (just not as bad because now the re-parsing happens only here and not in the regular handlers, so in the worst case only the WS RPC breaks).